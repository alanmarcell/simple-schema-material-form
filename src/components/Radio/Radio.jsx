import React from 'react';
import PropTypes from 'prop-types';
import { pathOr } from 'ramda';
import MaterialRadio, { RadioGroup } from 'material-ui/Radio';
import { compose, setDisplayName, pure, mapProps } from 'recompact';
import { FormControlLabel } from 'material-ui/Form';
import SelectionControl from '../SelectionControl';

const RadioSelectionControl = props => {
  const {
    onChange, label, selectOptions = [], value,
  } = props;

  return (
    <SelectionControl
      label={label}
    >
      <RadioGroup
        row
        aria-label={label}
        name={label}
        value={value}
        onChange={onChange}
      >
        {selectOptions.map(i =>
          (<FormControlLabel
            key={i.key}
            control={<MaterialRadio />}
            label={i.label}
            value={i.key}
          />))
        }
      </RadioGroup>
    </SelectionControl>
  );
};

const Radio = props => <RadioSelectionControl {...props} />;

/** This is our base enhancer to handle doc changes */
const enhance = compose(
  pure,
  setDisplayName('Radio'),
  mapProps(({
    fieldName, doc, setDoc, ...props
  }) => ({
    label: fieldName,
    value: pathOr('', [fieldName], doc),
    onChange: e => setDoc({ [fieldName]: e.target.value }),
    ...props,
  })),
);

const EnhancedRadio = enhance(RadioSelectionControl);

Radio.defaultProps = {
  value: '',
};

Radio.propTypes = {
  value: PropTypes.string,
  doc: PropTypes.object.isRequired,
  setDoc: PropTypes.func.isRequired,
};


export default EnhancedRadio;
