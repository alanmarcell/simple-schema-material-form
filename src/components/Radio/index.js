import RadioComponent from './Radio';

export default RadioComponent;
