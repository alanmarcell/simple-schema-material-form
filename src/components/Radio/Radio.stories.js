import React from 'react';
import { storiesOf } from '@storybook/react';
import { State, Store } from '@sambego/storybook-state';
import Radio from './';
import withTests from '../../stories/withTests';

const store = new Store({
  doc: {},
});

storiesOf('Fields|Radio', module)
  .addDecorator(withTests('Radio'))
  .add('simple', () => (
    <State store={store}>
      <Radio
        fieldName="test"
        setDoc={e => store.set({ doc: e })}
        selectOptions={[{ label: 1, key: 'um' }, { label: 2, key: 'dois' }]}
      />
    </State>
  ));
