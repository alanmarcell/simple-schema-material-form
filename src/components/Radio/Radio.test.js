import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Radio from './';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Radio fieldName="test" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow a Radio', () => {
  const setDoc = jest.fn();
  const RadioComponent = (<Radio
    fieldName="test"
    setDoc={setDoc}
    doc={{}}
    selectOptions={[{ label: 1, key: 'um' }, { label: 2, key: 'dois' }]}
  />);

  const component = shallow(RadioComponent);
  const RadioGroup = component.find('RadioGroup');

  RadioGroup.simulate('change', {
    target: { value: 'um' },
  });

  // Inspect doc prop field key
  expect(setDoc).toBeCalledWith({ test: 'um' });
  expect(setDoc).toHaveBeenLastCalledWith({ test: 'um' });
  expect(setDoc).toHaveBeenCalled();
  expect(component).toMatchSnapshot();
});

