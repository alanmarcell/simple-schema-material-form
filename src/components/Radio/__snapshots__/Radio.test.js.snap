// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow a Radio 1`] = `
<SelectionControl
  label="test"
>
  <RadioGroup
    aria-label="test"
    name="test"
    onChange={[Function]}
    row={true}
    value=""
  >
    <WithStyles(FormControlLabel)
      control={<WithStyles(Radio) />}
      key="um"
      label={1}
      value="um"
    />
    <WithStyles(FormControlLabel)
      control={<WithStyles(Radio) />}
      key="dois"
      label={2}
      value="dois"
    />
  </RadioGroup>
</SelectionControl>
`;
